doc/en/fixture.rst

```diff
@@ -115,32 +121,504 @@ for reference:
 .. _`@pytest.fixture`:
 .. _`pytest.fixture`:
 
-Fixtures as Function arguments
------------------------------------------
+What fixtures are
+-----------------
+
+Before we dive into what fixtures are, let's first look at what a test is.
+
+In the simplest terms, a test is meant to look at the result of a particular
+"behavior", and make sure that result aligns with what you would expect.
+Behavior is not something that can be empirically measured, which is why writing
```

Comment:
I see where you're coming from, and agree.

I'm thinking of something like logging in. Someone may want to test logging in, but there's no means of saying `assert user.logged_in is True` and have that statement be 100% correct. Instead, they may look at how the system responds to the context of a proper login request bring sent. They may check that they now have a cookie, or that they can access a resource only a logged in user should be able to access. But these still don't tell them if the user is logged in, because being logged in is an abstract concept we've attached to the way we want the code to behave given certain conditions.

Perhaps it would be better to say something like this?

> The intent or meaning behind how the state behaves is not something that can be empirically measured, which is why writing tests can be challenging.